sklearn/calibration.py

```diff
@@ -282,25 +290,38 @@ def fit(self, X, y, sample_weight=None, **fit_params):
         for sample_aligned_params in fit_params.values():
             check_consistent_length(y, sample_aligned_params)
 
-        if self.base_estimator is None:
+        if self.base_estimator != "deprecated":
+            if self.estimator is not None:
+                raise ValueError(
+                    "Both `base_estimator` and `estimator` are set. Only set "
+                    "`estimator` since `base_estimator` is deprecated."
+                )
+            warnings.warn(
+                "`base_estimator` was renamed to `estimator` in version 1.2 and "
+                "will be removed in 1.4.",
+                FutureWarning,
+            )
+            self.estimator = self.base_estimator
```

Comment:
This is usually an anti-pattern in scikit-learn. In this case, I think we we can get away with:

```python
if self.base_estimator != "deprecated":
    ...
    estimator = self.base_estimator
else:
    estimator = self.estimator

# Estimator from now on
if estimator is None:
    estimator = LinearSVC(random_state=0)

if self.cv == "prefit":
    check_is_fitted(estimator, ...)
```

WDYT @glemaitre ?

Reply:
I see what you mean. I think that you are right. Let me update it.